src/main/java/io/qdrant/spark/QdrantOptions.java

```diff
@@ -1,38 +1,79 @@
 package io.qdrant.spark;
 
 import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
 
 /** This class represents the options for connecting to a Qdrant instance. */
 public class QdrantOptions implements Serializable {
   public String qdrantUrl;
   public String apiKey;
   public String collectionName;
-  public String embeddingField;
   public String idField;
-  public String vectorName;
-  public int batchSize = 100;
+
+  public int batchSize = 64;
   public int retries = 3;
 
+  // Should've named the option 'vectorField'. But too late now.
+  public String embeddingField;
+  public String vectorName;
+
+  public String[] sparseVectorValueFields;
+  public String[] sparseVectorIndexFields;
+  public String[] sparseVectorNames;
+
+  public String[] vectorFields;
+  public String[] vectorNames;
+
+  public List<String> payloadFieldsToSkip = new ArrayList<String>();
+
   /**
    * Constructor for QdrantOptions.
    *
    * @param options A map of options for connecting to a Qdrant instance.
    */
   public QdrantOptions(Map<String, String> options) {
-    this.qdrantUrl = options.get("qdrant_url");
-    this.collectionName = options.get("collection_name");
-    this.embeddingField = options.get("embedding_field");
-    this.idField = options.get("id_field");
-    this.apiKey = options.get("api_key");
-    this.vectorName = options.get("vector_name");
+    qdrantUrl = options.get("qdrant_url");
+    collectionName = options.get("collection_name");
+    embeddingField = options.getOrDefault("embedding_field", "");
+    idField = options.getOrDefault("id_field", "");
+    apiKey = options.getOrDefault("api_key", "");
+    vectorName = options.getOrDefault("vector_name", "");
+    sparseVectorValueFields = options.getOrDefault("sparse_vector_value_fields", "").split(",");
+    sparseVectorIndexFields = options.getOrDefault("sparse_vector_index_fields", "").split(",");
+    sparseVectorNames = options.getOrDefault("sparse_vector_names", "").split(",");
+    vectorFields = options.getOrDefault("vector_fields", "").split(",");
+    vectorNames = options.getOrDefault("vector_names", "").split(",");
```

Comment:
if the col names can be used as the vector names we can reduce this double complexity.

the user will just have to do a col rename to achieve the result he expects.

Reply:
Wouldn't passing the vector name be simpler than enforcing the column to be same name as the vector?

Reply:
usually, developers including me. maintain the name of the vector as the col name. it also enforces better practices for the pipeline and makes the write options very intuitive.

I think what I'm trying to say is. to be able to map a different name to an existing col is a nice feature but it should not be compulsory.
let's say, I already have a correct name set for my cols for vector. I should be able to code
```python
df.write.option("vector_names","vector_1,vector_2,vector_3,vector_4").save()
```

instead of 
```python
df.write.option("vector_names","vector_1,vector_2,vector_3,vector_4").option("vector_fields","vector_1,vector_2,vector_3,vector_4").save()
```

Reply:
This is with the assumption that the vector names will be same as the column names.
But there have been column names like https://github.com/qdrant/qdrant-spark/issues/6#issuecomment-1951327998.

Reply:
agreed. now I'm convinced about the importance of renaming/name-mapping capability. 
 
would be great if we could reduce the complexity for already mapped col names.
something like, 
    if vector_fields are mentioned and vector_names are not, consider the names as vector_fields.
    (Of course, the states should be either null, or every field mapped (equal length), not in between)